client/my-sites/importer/importer-medium.jsx

```diff
@@ -12,34 +13,45 @@ import { localize } from 'i18n-calypso';
  */
 import FileImporter from './file-importer';
 
-/**
- * Module variables
- */
-const importerData = {
-	title: 'Medium',
-	icon: 'medium',
-};
-
 class ImporterMedium extends React.PureComponent {
 	static displayName = 'ImporterMedium';
 
-	render() {
-		importerData.description = this.props.translate(
-			'Import posts, tags, images, and videos ' + 'from a Medium export file.'
-		);
-
-		importerData.uploadDescription = this.props.translate(
-			'Upload a {{b}}Medium export file{{/b}} to start ' + 'importing into {{b2}}%(title)s{{/b2}}.',
+	static propTypes = {
+		site: PropTypes.shape( {
+			title: PropTypes.string.isRequired,
+		} ).isRequired,
+		importerStatus: PropTypes.shape( {
+			filename: PropTypes.string,
+			importerState: PropTypes.string.isRequired,
+			errorData: PropTypes.shape( {
+				type: PropTypes.string.isRequired,
+				description: PropTypes.string.isRequired,
+			} ),
+			percentComplete: PropTypes.number,
+			siteTitle: PropTypes.string.isRequired,
+			statusMessage: PropTypes.string,
+		} ),
+	};
+
+	importerData = {
+		title: 'Medium',
+		icon: 'medium',
+		description: this.props.translate(
+			'Import posts, tags, images, and videos from a Medium export file.'
+		),
```

Comment:
note that we typically leave translated strings in the `render()` path because otherwise they don't update when the language changes. that is, by setting `description` here it only gets set at the instantiation of the object and won't change into another language when it should